front/components/assistant/RenderMessageMarkdown.tsx

```diff
@@ -214,20 +227,33 @@ function isCiteProps(props: ReactMarkdownProps): props is ReactMarkdownProps & {
 
 function CiteBlock(props: ReactMarkdownProps) {
   const references = React.useContext(ReferencesContext);
+  const updateActiveReferences = React.useContext(
+    UpdateActiveReferencesContext
+  );
+  const refs =
+    isCiteProps(props) && props.references
+      ? (
+          JSON.parse(props.references) as {
+            counter: number;
+            ref: string;
+          }[]
+        ).filter((r) => r.ref in references)
+      : undefined;
+
+  useEffect(() => {
+    if (refs) {
+      refs.forEach((r) => {
+        const document = references[r.ref];
+        updateActiveReferences(document, r.counter);
```

Comment:
Should me marginally easier to send all knowns ref always that way you can get away with the logic to unify them in AgentMessage ?

Reply:
the cite block doesn't have access to all refs -- do you mean to compute them in the parent rendermarkdonw? (the references array has all docs even non-cited ones)

Reply:
each time you run that code you trigger `refs.length` events. I'm suggesting sending the array instead?

Reply:
:+1: sorry for the misunderstanding